src/main/java/dp/handler/Handler.java

```diff
@@ -353,12 +347,22 @@ private S3Object getObject(String bucket, String key, boolean isPublished) throw
     }
 
     private String getDownloadUrl(boolean isPublished, String filePath, WorkbookDetails details) {
-        if (isPublished && bucketUrl.length() > 0) {
+        if (isPublished && StringUtils.isNotEmpty(bucketUrl)) {
             return bucketUrl + "/" + filePath;
         }
         return details.getDownloadURI();
     }
 
+    public String getS3URL(String url) {
```

Comment:
Not sure this needs to be public as it very unlikely to ever be used by anything else plus its exposing the inner workings of the handler which isn't opitmal. If you make it package private
```
String getS3URL(String url) {
}
``` 
The unit test will still be able to call it but it will not visible to anything outside the package.

Reply:
Also consider adding a javadoc comment for this method - it wont be obvious why this is necessary to someone reading it for the first time. A short comment to give some context will be helpful.

Reply:
done